CPP-Example.cpp

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+   char src[40];
+   char dest[100];
+   char dest2[100];
+
+   gets(src); 
+
+   strcpy(dest, src); 
+ 
+   strncpy(dest,src, strlen(dest)); //DevSkim: ignore DS111237,DS154189 
+   
+   strcat(dest, "this is bad");
+
+   strncat(dest, "this is also bad", strlen(dest)); 
```

Comment:
[Banned C function detected (strncat)](https://github.com/microsoft/devskim/guidance/DS108330.md)
strncat adds the null terminator at character 'n + 1', rather than at the nth character. this frequently leads to the null terminator being added in the memory adjacent to the destination buffer, rather than in the destination buffer.

Replacement: Use strcat_s or strlcat if possible. 


Suggestions:
**[1]** *+   strcat_s(dest,  strlen(dest,  "this is also bad")); *
**[2]** *+   strlcat(dest,  "this is also bad",  strlen(dest)); *